types/joi/index.d.ts

```diff
@@ -825,20 +844,26 @@ export interface Terms {
 export interface Rules {
     name: string;
     params?: ObjectSchema | { [key: string]: Schema };
-    setup?: Function;
-    validate?: Function;
+    setup?: (this: AnySchema<AnySchema<Schema>>, description: { [key: string]: any }) => AnySchema<AnySchema<Schema>> | void;
```

Comment:
This is actually incorrect, this should `Err | any` as you are meant to return the value.

Unfortunately this passed inside the test as `value` was treated as `any` already.

We should probably create a follow-up PR soon-ish to avoid people getting confused in case they type `value` manually...

I can do this tonight unless you want to grab this one.